### PR TITLE
[Report Page] Fix bugs related to categories

### PR DESCRIPTION
**Pull request by j-x-han**
# Description

Fixing https://jira.czi.team/browse/IDSEQ-1950 and https://jira.czi.team/browse/IDSEQ-1951:

* Taxon tree will no longer crash if no taxons pass the selected filters (the tree will no longer be rendered unless there are taxons).
image

* Uncagtegorized taxons will now show up if you select that filter.
image

# Tests

* Applied filters until no taxons passed, selected taxon tree view, verified that page did not crash.
* Applied the "uncategorized" category filter and verified that the results matched that of the legacy page.

### CODE REVIEW

File: app/assets/src/components/views/SampleViewV2/SampleViewV2.jsx [hunk 1]
```diff
@@ -332,11 +332,12 @@ export default class SampleViewV2 extends React.Component {
 
     // taxon's category was selected and its subcategories were not excluded
```
Comment (gregdingle) [inline app/assets/src/components/views/SampleViewV2/SampleViewV2.jsx]: so categories is filter by inclusion and subcategories by exclusion? seems confusing!

File: app/assets/src/components/views/SampleViewV2/SampleViewV2.jsx [hunk 2]
```diff
@@ -1029,19 +1030,20 @@ export default class SampleViewV2 extends React.Component {
...
-                nameType={selectedOptions.nameType}
-                onTaxonClick={this.handleTaxonClick}
-                sample={sample}
-                taxa={filteredReportData}
-                useReportV2Format={true}
-              />
-            </div>
-          )}
+          {view == "tree" &&
+            filteredReportData.length > 0 && (
```
Comment (gregdingle) [inline app/assets/src/components/views/SampleViewV2/SampleViewV2.jsx]: did we consider adding an empty state message to the user such as "no data passed the selected filters" ?

Comment (j-x-han) [reply app/assets/src/components/views/SampleViewV2/SampleViewV2.jsx]: There's a small line underneath the filters that tells the user how many rows are passing the selected filters, which will say "0 rows passing the above filters, out of # total rows."

File: app/assets/src/components/views/SampleViewV2/SampleViewV2.jsx (hunk 1 above)
Comment (j-x-han) [reply app/assets/src/components/views/SampleViewV2/SampleViewV2.jsx]: I think subcategories are also filtered by inclusion; there's a condition just above this one that explicitly checks the subcategory.

**Pull request merged**